portal/src/main/webapp/js/src/oncoprint/setup.js

```diff
@@ -2499,9 +2499,153 @@ window.CreateCBioPortalOncoprintWithToolbar = function (ctr_selector, toolbar_se
 		    cbio.download.initDownload(content, downloadOpts);
 		});
 	    });
+	    
+	    $('body').on('click', '.oncoprint-tabular-download', function () {
+		//Get caseNames (either patients or samples)
+		window.QuerySession.getUIDToCaseMap().then(function (uid_to_case) {
+		    var caseNames = [];
+		    oncoprint.getIdOrder().map(function (uid) {
+			caseNames.push(uid_to_case[uid]);
```

Comment:
just want to confirm `map` always goes in order

Reply:
> `map` calls a provided `callback` function **once for each element** in an array, in order, and constructs a new array from the results.

From: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map